src/edu/stanford/nlp/sempre/ScopedFormula.java

```diff
@@ -0,0 +1,71 @@
+package edu.stanford.nlp.sempre;
+
+import java.util.*;
+
+import com.google.common.base.Function;
+
+import fig.basic.LispTree;
+
+/**
+ * Represent a binary with a restrict domain (scope).
```

Comment:
It's only used in my LambdaDCSExecutor. It might be better if I put this (and ScopedValue / InfiniteListValue) in the tables package as they are not core types of formulas / values.